src/main/scala/chiseltest/legacy/backends/verilator/VerilatorExecutive.scala

```diff
@@ -48,6 +55,115 @@ object VerilatorExecutive extends BackendExecutive {
     val circuit = elaboratedAnno.collect { case x: ChiselCircuitAnnotation => x }.head.circuit
     val dut = getTopModule(circuit).asInstanceOf[T]
 
+    if (elaboratedAnno.contains(CachingAnnotation)) {
```

Comment:
Since the whole thing has gotten quite long, could you please move this into its own function?

Reply:
Moved this into `getCachedSim`. Moved caching generation into `cacheSim`.